### PR TITLE
Add cKeys' "The Dora" Board

### PR DESCRIPTION
**Pull request by brandenbyers**
## Description
Added the cKeys 4x5 keyboard with rotary encoder and Proton C pinouts that coincides with cKeys' soldering workshops. The default keymap is used as a teaching tool and introduction to QMK.

## Types of changes

- [ ] Core
- [ ] Bugfix
- [ ] New Feature
- [ ] Enhancement/Optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/Layout/Userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by this PR
*  N/A

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document. (https://docs.qmk.fm/#/contributing)
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by brandenbyers** [comment]
This is one of the keyboards that was just debuted on Sunday at the SMKmeetup workshops. Nearly 30 people attended the workshop. I plan to create a lot of future content around this board to help teach people about different aspects of QMK. For example, I am currently working on content for experimenting with LED matrices using the pinouts on this board along with a breadboard.

File: keyboards/ckeys/thedora/config.h [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+/* COL2ROW, ROW2COL, or CUSTOM_MATRIX */
+#define DIODE_DIRECTION COL2ROW
+
+//#define BACKLIGHT_PIN B7
+// #define BACKLIGHT_BREATHING
+//#define BACKLIGHT_LEVELS 3
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCING_DELAY 5
```
Comment (fauxpark) [inline keyboards/ckeys/thedora/config.h]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/ckeys/thedora/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+//           │    1    │    2    │    3    │    -    │  MIDI   │
+//           │         │         │         │         │         │
+// ┌─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
+// │  TOGGLE │         │         │         │         │         │
+// │  MUSIC  │    0    │    .    │    =    │    +    │  MUSIC  │
+// │         │         │         │         │         │         │
+// └─────────┴─────────┴─────────┴─────────┴─────────┴─────────┘
+
+  [_BASE] = LAYOUT(
+    KC_KP_7, KC_KP_8,   KC_KP_9,     KC_PSLS, MO(_ADMIN), MU_TOG, \
```
Comment (fauxpark) [inline keyboards/ckeys/thedora/keymaps/default/keymap.c]: The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater without them.

File: keyboards/ckeys/thedora/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+enum layers {
+  _BASE,                // base layer
+  _MUSIC,               // music mode
+  _MIDI,                // midi mode
+  _MOUSE,               // mouse keys
+  _ADMIN                // admin duties
+};
+
+extern keymap_config_t keymap_config;
```
Comment (fauxpark) [inline keyboards/ckeys/thedora/keymaps/default/keymap.c]:
```suggestion
```

File: keyboards/ckeys/thedora/keymaps/default/keymap.c (hunk 2 above)
Comment (brandenbyers) [reply keyboards/ckeys/thedora/keymaps/default/keymap.c]: I had no idea I could remove those. I just kept doing what I've always been doing for the past two years. I agree; much neater without. Removing...

**Comment by brandenbyers** [comment]
Any tips on fixing the moxygen `linebreak` and `simplesectsep` failures in Travis? The only place that I'm seeing `simplesectsep` is in `lib/lufa/LUFA/StudioIntegration/HV1/lufa_docbook_transform.xsit`:

```
	<xsl:template match="linebreak | simplesectsep">
		<!-- MUST be on two separate lines, as this is a *literal* newline -->
		<literallayout>
		</literallayout>
	</xsl:template
```

I'm having a hard time understanding how any of my keymap additions would have affected this. Any help much appreciated! Thanks!

**Comment by fauxpark** [comment]
The docs build failure (?) is for the most part uninteresting, I think. It seems to be a side effect of the actual firmware build failing - if you scroll up, you'll see it's complaining that `NUMBER_OF_ENCODERS` is being redefined. QMK actually calculates this automatically as of a couple of weeks ago, based on the number of encoder pins declared.

File: keyboards/ckeys/thedora/config.h [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0xFEED
+#define PRODUCT_ID 0x6060
+#define DEVICE_VER 0x0001
+#define MANUFACTURER ckeys
+#define PRODUCT thedora
+#define DESCRIPTION A board for keyboard exploration.
+
+#define NUMBER_OF_ENCODERS 1
```
Comment (fauxpark) [inline keyboards/ckeys/thedora/config.h]:
```suggestion
```

**Comment by brandenbyers** [comment]
Thanks @fauxpark; suggestion committed. However, when that line is removed, I get errors when trying to make the keyboard:

```
5:4: error: #error "Number of encoders not defined by NUMBER_OF_ENCODERS"
   #error "Number of encoders not defined by NUMBER_OF_ENCODERS"
    ^~~~~
quantum/encoder.c:32:29: error: 'NUMBER_OF_ENCODERS' undeclared here (not in a function); did you mean 'NUMBER_OF_MODES'?
 static pin_t encoders_pad_a[NUMBER_OF_ENCODERS] = ENCODERS_PAD_A;
                             ^~~~~~~~~~~~~~~~~~
                             NUMBER_OF_MODES
quantum/encoder.c:38:15: error: 'encoder_value' defined but not used [-Werror=unused-variable]
 static int8_t encoder_value[NUMBER_OF_ENCODERS] = {0};
               ^~~~~~~~~~~~~
quantum/encoder.c:37:16: error: 'encoder_state' defined but not used [-Werror=unused-variable]
 static uint8_t encoder_state[NUMBER_OF_ENCODERS] = {0};
                ^~~~~~~~~~~~~
quantum/encoder.c:33:14: error: 'encoders_pad_b' defined but not used [-Werror=unused-variable]
 static pin_t encoders_pad_b[NUMBER_OF_ENCODERS] = ENCODERS_PAD_B;
              ^~~~~~~~~~~~~~
quantum/encoder.c:32:14: error: 'encoders_pad_a' defined but not used [-Werror=unused-variable]
 static pin_t encoders_pad_a[NUMBER_OF_ENCODERS] = ENCODERS_PAD_A;
              ^~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

Hence the reason why I added the number of encoders bit originally. Is there anything else that I need to add to the `config.h` besides these three lines to get the encoder to work with the latest QMK encoder updates?

```
define ENCODERS_PAD_A { B13 }
#define ENCODERS_PAD_B { B15 }
#define ENCODER_RESOLUTION 4
```

**Comment by fauxpark** [comment]
Ah, your fork is lagging behind upstream by quite a bit, on both the `exBoard` and master branches - so it probably doesn't have the changes in that allows `NUMBER_OF_ENCODERS` to be calculated. This also explains why Travis is building a whole bunch of boards that you haven't touched. In any case, it is not complaining now.

**Comment by brandenbyers** [comment]
# 🤦️

Now I feel silly forgetting to pull upstream. Lol. Thanks for your help!

File: keyboards/ckeys/thedora/readme.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+# The Dora Board
+
+This keyboard was built for QMK exploration big and small. What will you discover next?
+
+![The Dora Board](https://ckeys.org/images/exboard-3.jpg)
+
+## Slides
```
[the remark this answers is not in the document]
Comment (brandenbyers) [reply keyboards/ckeys/thedora/readme.md]: Can do! For clarification, can I add the extra info after the boilerplate? Or should I put that info outside of the repo?

Also, can I update our other keyboard readmes in one separate PR? Or should I do separate PRs for each keyboard readme boilerplate update?

File: keyboards/ckeys/thedora/config.h [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+//#define MATRIX_HAS_GHOST
+
+/* number of backlight levels */
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+#define AUDIO_CLICKY
```
Comment (drashna) [inline keyboards/ckeys/thedora/config.h]: ❤️

File: keyboards/ckeys/thedora/config.h [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+//#define MAGIC_KEY_LAYER8         8
+//#define MAGIC_KEY_LAYER9         9
+//#define MAGIC_KEY_BOOTLOADER     PAUSE
+//#define MAGIC_KEY_LOCK           CAPS
+//#define MAGIC_KEY_EEPROM         E
+//#define MAGIC_KEY_NKRO           N
+//#define MAGIC_KEY_SLEEP_LED      Z
+
+// Audio Click
+#define AUDIO_CLICKY
```
Comment (drashna) [inline keyboards/ckeys/thedora/config.h]: As much as I love Audio Clicky, this is defined twice!  :D 

This instance should be removed: 
```suggestion
```

**Pull request merged**